RELEASE/scripts/sl_ascend/sl_cooking.ash

```diff
@@ -1582,37 +1592,163 @@ void consumeStuff()
+
+	float desirability; // adv count that will be used for optimization
+	                    // (lower for pulls, higher for buffs/tower keys)
+
+	int organ;          // SL_ORGAN_*
+	int howToGet;       // SL_OBTAIN_*
+};
+
+string consumable_name(ConsumeAction action)
+{
+	string name = "<name not found>";
+	if (action.it != $item[none]) name = to_string(action.it);
+	else if (action.organ == SL_ORGAN_LIVER) name = cafeDrinkName(action.cafeId);
+	else if (action.organ == SL_ORGAN_STOMACH) name = cafeFoodName(action.cafeId);
+	return name;
+}
+
+string to_string(ConsumeAction action)
+{
+	string ret = "";
+	ret += "ConsumeAction(it="+action.it;
+	ret += ",cafeId="+action.cafeId;
+	ret += ",size="+action.size;
+	ret += ",adventures="+action.adventures;
+	ret += ",desirability="+action.desirability;
+	ret += ",organ="+action.organ;
+	ret += ",howToGet="+action.howToGet;
+	ret += ")";
+	return ret;
+}
+
+ConsumeAction MakeConsumeAction(item it)
+{
+	int organ = it.inebriety > 0 ? SL_ORGAN_LIVER : SL_ORGAN_STOMACH;
+	int size = max(it.inebriety, it.fullness);
+	float adv = expectedAdventuresFrom(it);
+	return new ConsumeAction(it, 0, size, adv, adv, organ, SL_OBTAIN_NULL);
+}
+
+boolean slPrepConsume(ConsumeAction action)
+{
+	print(to_string(action));
+	if(action.howToGet == SL_OBTAIN_PULL)
+	{
+		print("slPrepConsume: Pulling a " + action.it, "blue");
+		action.howToGet = SL_OBTAIN_NULL;
+		return pullXWhenHaveY(action.it, 1, item_amount(action.it));
+	}
+	else if(action.howToGet == SL_OBTAIN_CRAFT)
+	{
+		print("slPrepConsume: Crafting a " + action.it, "blue");
+		action.howToGet = SL_OBTAIN_NULL;
+		return create(1, action.it);
+	}
+	else if (action.howToGet == SL_OBTAIN_NULL)
+	{
+		print("slPrepConsume: Doing nothing to get a " + action.it, "blue");
+	}
+	return true;
+}
+
+boolean slConsume(ConsumeAction action)
+{
+	if (action.howToGet == SL_OBTAIN_PULL || action.howToGet == SL_OBTAIN_CRAFT)
+	{
+		abort("ConsumeAction not prepped: " + to_string(action));
+	}
+
+	if (action.organ == SL_ORGAN_LIVER)
+	{
+		buffMaintain($effect[Ode to Booze], 20, 1, action.size);
+	}
+	if(action.cafeId != 0)
+	{
+		if (action.organ == SL_ORGAN_LIVER)
+		{
+			return slDrinkCafe(1, action.cafeId);
+		}
+		else if (action.organ == SL_ORGAN_STOMACH)
+		{
+			return slEatCafe(1, action.cafeId);
+		}
+	}
+	else if(action.it != $item[none])
+	{
+		if (action.organ == SL_ORGAN_LIVER)
+		{
+			return slDrink(1, action.it);
+		}
+		else if (action.organ == SL_ORGAN_STOMACH)
+		{
+			return slEat(1, action.it);
+		}
+		else
+		{
+			abort("slConsume: Unrecognized organ " + action.organ);
+		}
+	}
+	abort("slConsume: exited with nothing");
+	return false;
+}
+
+boolean loadConsumables(string _type, ConsumeAction[int] actions)
 {
 	// Just in case!
 	if(sl_my_path() == "Dark Gyffte")
 	{
 		abort("We shouldn't be calling loadConsumables() in Dark Gyffte. Please report this.");
 	}
 
+	if (item_amount($item[unremarkable duffel bag]) > 0)
```

Comment:
Maybe only do this in ronin/hardcore, because testing out the knapsack based consumption in aftercore once used up all my stockpiled duffel bags and van keys, which was a mild bummer.

Reply:
WHOOPS yes